src/bindgen/config.rs

```diff
@@ -42,18 +44,48 @@ impl FromStr for Language {
             "C" => Ok(Language::C),
             "cython" => Ok(Language::Cython),
             "Cython" => Ok(Language::Cython),
+            "Java-JNA" => Ok(Language::JavaJna),
+            "Java-Jna" => Ok(Language::JavaJna),
+            "java-jna" => Ok(Language::JavaJna),
+            "JavaJNA" => Ok(Language::JavaJna),
+            "JavaJna" => Ok(Language::JavaJna),
+            "javajna" => Ok(Language::JavaJna),
             _ => Err(format!("Unrecognized Language: '{}'.", s)),
         }
     }
 }
 
+impl Display for Language {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Language::Cxx => {
+                    "cxx"
+                }
+                Language::C => {
+                    "c"
+                }
+                Language::Cython => {
+                    "cython"
+                }
+                Language::JavaJna => {
+                    "java-jna"
+                }
+            }
+        )
+    }
+}
+
 deserialize_enum_str!(Language);
 
 impl Language {
     pub(crate) fn typedef(self) -> &'static str {
         match self {
             Language::Cxx | Language::C => "typedef",
             Language::Cython => "ctypedef",
+            _ => unimplemented!(),
```

Comment:
Let's list JNA explicitly. But why can't it hit it?

Reply:
this typedef function is only used by the enum code